File: src/my_cpp_pkg/src/robot_news_station.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

class RobotNewsStationNode : public rclcpp::Node
{
public:
    RobotNewsStationNode() : Node("robot_news_station")
    {
        this->declare_parameter("robot_name", "JOHNNY_5");
        this->declare_parameter("publish_frequency", 500);

        robot_name_ = this->get_parameter("robot_name").as_string();
        int publish_frequency_ = this->get_parameter("publish_frequency").as_int();

        publisher_ = this->create_publisher<example_interfaces::msg::String>("robot_news", 10);
        timer_ = this->create_wall_timer(std::chrono::milliseconds((int)(1000.0 / publish_frequency_)), std::bind(&RobotNewsStationNode::publishNews, this));
        RCLCPP_INFO(this->get_logger(), "Robot news station has been started");
    }

private:
    void publishNews()
    {
        auto msg = example_interfaces::msg::String();
        msg.data = std::string("Hello, this is ") + robot_name_ + std::string(" from the Robot news station");
        publisher_->publish(msg);
    }
    std::string robot_name_;
    rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<RobotNewsStationNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/robot_news_station.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String
    
class RobotNewsStation(Node): 
    def __init__(self):

        super().__init__("robot_news_station") 

  
        self.declare_parameter("robot_name_param", "default_robot_name")
        self.robot_name_ = self.get_parameter("robot_name_param").value

        # create a publisher with string msg type, the topic called robot_news 
        # and with a duration of 10 seconds before timing out.
        self.publisher_ = self.create_publisher(String, "robot_news", 10)

        # create a timer to publish onto the topic as well and give it a callback
        self.timer_ = self.create_timer(0.5, self.publish_news)

        # put in a logger to make sure the node is working
        self.get_logger().info("Robot news station has been started.")

    # publish the message ie. Hello onto the publisher created above 
    def publish_news(self):
        msg = String()
        msg.data = "Hi this is " + self.robot_name_ + " from the robot news station"
        self.publisher_.publish(msg)
     
def main(args=None):
    rclpy.init(args=args)
    node = RobotNewsStation()
    rclpy.spin(node)
    rclpy.shutdown()
    
    
if __name__ == "__main__":
    main()